feat: Add survey response usage widget for non-admin users

nehaaprasad opened the pull request:
### I made a Pull request!

## Problem

- Non-admin users can't see their survey response usage limits and have to ask admins to check this info. This creates unnecessary friction since users need to know how many responses they have left (250/month free limit).


#35160


## Changes

- Added `usage_summary` to `BillingType` interface
- Created `SurveyResponseLimitWidget` component showing usage vs limit for non-admin users
- Integrated widget into Surveys page (Active/Archived tabs)
- Connected billing logic to surveys logic
- Added test coverage

## How did you test this code?

- Unit tests for all widget states
- Type checking verification
- Integration testing with surveys page
- Manual testing for non-admin visibility

## Feature Requirements Met:

- Display how many survey responses I have seen this month  - Shows current usage
- How much left I have for free  - Shows remaining responses
- Only admins can see how many surveys responses there are left  - Widget only shows for non-admins
- Widget in the surveys page  - Positioned correctly in surveys page
- Even as a non-admin  - Specifically targets non-admin users


## Technical Implementation:

- Data Source: Uses existing billing API (/api/billing)
- State Management: Uses Kea logic properly
- UI Components: Uses PostHog's LemonBanner, Tooltip
- TypeScript: Proper typing throughout
- Testing: Comprehensive test coverage
- Performance: Only loads billing data when needed


<img width="1273" height="480" alt="image" src="https://github.com/user-attachments/assets/e9c05248-4530-404e-853b-cae76c32986e" />
<img width="1273" height="480" alt="image" src="https://github.com/user-attachments/assets/c2e5248c-50a2-4c74-885c-a044c1e2878e" />
<img width="1273" height="480" alt="image" src="https://github.com/user-attachments/assets/e1806bdc-ecbe-4244-b513-753db1743f72" />

nehaaprasad committed:
frontend/src/scenes/surveys

lucasheriques committed:
Merge branch 'master' into feat/filtering-survey-answers

nehaaprasad committed:
Merge branch 'master' into feat/filtering-survey-answers

nehaaprasad committed:
survey filters now only show categories that make sense for survey analysis

nehaaprasad committed:
Added missing interface field, built the widget with proper logic, integrated it into the surveys page, connected billing logic, and added full test coverage.

nehaaprasad committed:
Merge branch 'master' into survey-responses

nehaaprasad committed:
update some issue

nehaaprasad committed:
Merge branch 'master' into survey-responses

nehaaprasad committed:
Merge branch 'master' into survey-responses

lucasheriques commented on frontend/src/scenes/surveys/SurveyResponseLimitWidget.tsx:
```diff
@@ -0,0 +1,64 @@
+    const { user } = useValues(userLogic)
+
+    // Only show for non-admin users
+    if (user?.is_staff || user?.is_impersonated) {
```
`is_staff` and `is_impersonated` are not for user roles, but instead, Django attributes to determine wether the current user has access to the admin panel or if they are impersonating someone

for checking if they are an admin, you can instead import the organization logic like this:

`const { isAdminOrOwner } = useValues(organizationLogic)`

there are some examples in the codebase

lucasheriques commented on frontend/src/scenes/surveys/SurveyResponseLimitWidget.tsx:
```diff
@@ -0,0 +1,64 @@
+
+    // Check if we have survey response usage data
+    const surveyResponsesUsage = billing?.usage_summary?.survey_responses
+    if (!surveyResponsesUsage) {
```
```suggestion
    if (!surveyResponsesUsage.usage) {
```

if usage is 0 then we can just hide it

lucasheriques commented on frontend/src/scenes/surveys/SurveyResponseLimitWidget.tsx:
```diff
@@ -0,0 +1,64 @@
+        return null
+    }
+
+    const { usage = 0, limit = DEFAULT_SURVEY_RESPONSE_LIMIT } = surveyResponsesUsage
```
don't use a default value for limit. if the account doesn't have one, let's just not use it, and instead just show how many responses there are

lucasheriques commented on frontend/src/scenes/surveys/SurveyResponseLimitWidget.tsx:
```diff
@@ -0,0 +1,64 @@
+                        <IconInfo className="text-muted" />
+                    </Tooltip>
+                </div>
+                {percentageUsed >= 80 && (
+                    <Link to="/organization/billing" className="font-medium underline">
+                        upgrading your plan
+                    </Link>
+                )}
```
link to billing should be to admins only

lucasheriques reviewed (commented):
hey @naaa760 thank you for your second PR! 

Curious: where are those screenshots from? They don't see to be seen from the PostHog dashboard.

Also, I actually like the ability to just show how many responses we have collected in the current billing period. I think we should change it to display it to all users, just hide it when impersonating.

If usage is 0, let's just not show the limit too.

Here's my though process:

- I actually think it's useful for any survey user to see how many responses they have collected this month across all surveys
- However, only admins should be able to see what's the limit and the status changes according to that

i left a few other comments on other parts of the code but overall I really like this and appreciate you doing it! if you make those changes, we'll get this in ASAP

nehaaprasad committed:
Merge branch 'master' into survey-responses

nehaaprasad committed:
Fix reviewer feedback: use isAdminOrOwner, show for all users except impersonated, remove default limit, fix billing link visibility

naaa760 commented:
Hi @lucasheriques, sir thanks for the feedback! I've fixed all the issues:

-  Changed to use `isAdminOrOwner` from `organizationLogic` instead of `is_staff`
- Fixed usage check to `if (!surveyResponsesUsage?.usage)` 
- Removed default limit value - now shows usage without limit if account doesn't have one
- Billing link only shows for admin/owner users
- Widget shows for all users except when impersonating

let me know next things!

lucasheriques commented on frontend/src/scenes/surveys/SurveyResponseLimitWidget.tsx:
```diff
@@ -3,62 +3,61 @@ import { LemonBanner, Link, Tooltip } from '@posthog/lemon-ui'
+    let type: 'info' | 'warning' | 'error' = 'info'
+    let message: string
+
+    if (usage === 0) {
+        message = `You have used 0 responses this month.`
+    } else if (limit && usage >= limit) {
+        type = 'error'
+        message = `You have used all ${limit} responses this month. Further responses will not be collected.`
+    } else if (limit && percentageUsed >= 80) {
+        type = 'warning'
+        message = `You have used ${usage} of ${limit} responses this month. You have ${
+            limit - usage
+        } responses remaining.`
+    } else if (limit) {
+        message = `You have used ${usage} of ${limit} responses this month. You have ${
+            limit - usage
+        } responses remaining.`
+    } else {
+        message = `You have used ${usage} responses this month.`
```
I think the language here can be improved. Think about the UX people seeing this: `used X responses` doesn't really make a lot of sense for the surveys page.

Instead, we should do something like `you have received X responses this month. Your current limit is Y responses per month.`

we don't really have to use the billing-side of language, since I think we have two goals here:

- People should know how many responses they received
- They should also know how many more they can still receive until they reach their limit

what do you think?

lucasheriques reviewed (commented):
one last comment, let me know what you think

nehaaprasad committed:
feat: improve survey response limit widget messaging

lucasheriques reviewed (commented):
last couple of comments, but overall looks good! once you address them and fix the CI issues we can get this merged

nehaaprasad committed:
Fix unintended removals and redundant logic in surveys code

- Restore missing values and actions in surveysLogic connect block:
  - userLogic.hasAvailableFeature
  - teamLogic.currentTeam, currentTeamLc
  - teamLogic.loadCurrentTeam, addProductIntent
- Fix redundant early return check in SurveyResponseLimitWidget:
  - Remove unnecessary .usage check since we already check if surveyResponsesUsage exists

nehaaprasad committed:
Merge branch 'master' into survey-responses

naaa760 commented:
@lucasheriques 
please review the changes of this PR.

nehaaprasad committed:
Merge branch 'master' into survey-responses

lucasheriques reviewed (commented):
there's still some comments that need addressing, can you double check? regarding when the usage is 0, plus `VersionCheckerBanner` being removed
